File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/places/DeletePlaceHandler.java
package by.fpmibsu.findafriend.controller.commands.places;

import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.place.PlaceDao;

public class DeletePlaceHandler extends RequestHandler<Boolean, DeletePlaceCommand> {
    private final PlaceDao placeDao;

    public DeletePlaceHandler(PlaceDao placeDao) {
        this.placeDao = placeDao;
    }

    @Override
    public Boolean handle(DeletePlaceCommand request) {
        return placeDao.delete(request.getPlaceId());
    }
}

File: BackendNET/Backend/Commands/Places/DeletePlace/DeletePlaceHandler.cs
using Backend.Repository;
using MediatR;

namespace Backend.Commands.Places.DeletePlace;

public class DeletePlaceHandler : IRequestHandler<DeletePlaceCommand, bool>
{
    private readonly IPlaceRepository _placeRepository;

    public DeletePlaceHandler(IPlaceRepository placeRepository)
    {
        _placeRepository = placeRepository;
    }

    public async Task<bool> Handle(DeletePlaceCommand request, CancellationToken cancellationToken)
    {
        return await _placeRepository.DeleteAsync(request.PlaceId, cancellationToken) == 1;
    }
}